assets/js/react/GamePlayer/components/CommandInput.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from "react"
+import PropTypes from "proptypes"
+
+class CommandInput extends Component {
+  static propTypes = {
+    player: PropTypes.object.isRequired,
+    socket: PropTypes.object.isRequired,
+  }
+
+  constructor(props) {
+    super(props)
+    this.input = React.createRef()
+
+    const { player, socket } = this.props
+    const channel = socket.channel(`player:${player.playerId}`)
+    channel.join()
+    this.channel = channel
```

Comment:
as mentioned below, this construct feels a little weird.  There is probably a better way to share the socket/channel connection.  but i want to talk about it before i make another mod.  This is currently working.

Here we do `channel.join` and if we do it in both places, js complains that you can't double join.  So it might be better to handle that `join` in one spot.

Reply:
To share the channel/socket, what about a `Channel` component or something that lives just above `GamePlayer`?  It could do the initial creation of the socket, and then when the player joins, it could call some function in `Channel` to do the handshake.  Then components needing the channel could use a HOC to inject the channel or socket as a prop.

I'm new to React so I have no idea really.  Or you could just hold onto the channel/socket in a module in a variable.